staging/src/k8s.io/component-base/logs/json/json.go

```diff
@@ -78,18 +97,19 @@ func (l *zapLogger) handleFields(args []interface{}, additional ...zap.Field) []
 	// a slightly modified version of zap.SugaredLogger.sweetenFields
 	if len(args) == 0 {
 		// fast-return if we have no suggared fields.
-		return additional
+		return append(additional, zap.Any("v", l.lvl))
 	}
 
 	// unlike Zap, we can be pretty sure users aren't passing structured
 	// fields (since logr has no concept of that), so guess that we need a
 	// little less space.
-	fields := make([]zap.Field, 0, len(args)/2+len(additional))
+	fields := make([]zap.Field, 0, len(args)/2+len(additional)+1)
+	fields = append(fields, zap.Any("v", l.lvl))
 	for i := 0; i < len(args)-1; i += 2 {
 		// check just in case for strongly-typed Zap fields, which is illegal (since
 		// it breaks implementation agnosticism), so we can give a better error message.
 		if _, ok := args[i].(zap.Field); ok {
-			l.dPanic("strongly-typed Zap Field passed to logr", zap.Any("zap field", args[i]))
+			l.Info("strongly-typed Zap Field passed to logr")
```

Comment:
nit: if possible we should try to avoid using Info to avoid recursion by mistake. If possible we should call zap directly with simpler function.